community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -366,14 +365,15 @@ public Object nodeGetProperty( long nodeId, int propertyKeyId ) throws EntityNot
     }
 
     @Override
-    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int... relTypes )
+    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int[] relTypes )
             throws EntityNotFoundException
     {
         statement.assertOpen();
         try ( Cursor<NodeItem> node = dataRead().nodeCursorById( statement, nodeId ) )
         {
-            return new CursorRelationshipIterator(
-                    node.get().relationships( direction( direction ), deduplicate( relTypes ) ) );
+            return new CursorRelationshipIterator( dataRead()
+                    .nodeGetRelationships( statement, node.get(), direction( direction ),
+                            asSet( relTypes, t -> t >= 0 ) ) );
```

Comment:
So it'll silently filter negative ids, why do we want to hide such illegal arguments?

Have you measured this change going from int[] to PrimitiveIntSet? Quick comparison locally shows at least 10x slower, on a very low level. Would be great to measure this first.

Also later on we'd like to pass in sorted type ids to be able to take advantage of the store format better and this would go against that, I feel.

This comment applies to the execution stack beyond this point of course as well.